api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -46,8 +47,10 @@ private Types() {}
           .put(DoubleType.get().toString(), DoubleType.get())
           .put(DateType.get().toString(), DateType.get())
           .put(TimeType.get().toString(), TimeType.get())
-          .put(TimestampType.withZone().toString(), TimestampType.withZone())
-          .put(TimestampType.withoutZone().toString(), TimestampType.withoutZone())
+          .put(TimestampType.microsWithZone().toString(), TimestampType.microsWithZone())
+          .put(TimestampType.microsWithoutZone().toString(), TimestampType.microsWithoutZone())
```

Comment:
I don't think that the original lines need to change. We just need to add nanoseconds. Otherwise the change is lager than it needs to be.

Reply:
Are you just suggesting we leave lines 49 and 50 alone, i.e. using the methods we want to deprecate? Functionally there's no difference, but I'd prefer not to use the deprecated forms.

Reply:
Not using deprecated methods is a fair reason to push back, but I think it makes more sense to deprecate in the next PR so that we can evaluate the addition in isolation.

Reply:
Done.